### PR TITLE
Improvement: Add helper for normalizing issuer URI

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**

### CODE REVIEW

File: api/src/main/java/com/okta/idx/sdk/api/util/ClientUtil.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
```
Comment (JayNewstrom) [inline api/src/main/java/com/okta/idx/sdk/api/util/ClientUtil.java:2]: 2021

File: api/src/main/java/com/okta/idx/sdk/api/util/ClientUtil.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+     * @return the normalized full url
+     */
+    public static String getNormalizedUri(String issuer, String resourceUri) throws MalformedURLException {
+        String normalizedUri;
+
+        if (isRootOrgIssuer(issuer)) {
+            // remove trailing / before concatenating with /oauth2
+            normalizedUri = issuer.replaceAll("$/", "") + "/oauth2" + resourceUri;
+        } else {
+            normalizedUri = issuer + resourceUri;
```
Comment (JayNewstrom) [inline api/src/main/java/com/okta/idx/sdk/api/util/ClientUtil.java:43]: Should we replace the trailing slash on issuer here as well?

File: impl/src/test/groovy/com/okta/idx/sdk/impl/util/TestClientUtil.groovy [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    void issuerUri_nonRootOrgTest() {
+        assertThat "issuer uri expected to be non-root/org)",
+                !isRootOrgIssuer("https://sample.okta.com/oauth2/default")
+        assertThat "issuer uri expected to be non-root/org)",
+                !isRootOrgIssuer("https://example.io/oauth2/ausvd5ple5TRRsbcJ0h7")
+    }
+
+    @Test
+    void normalizedUri_rootOrgTest() {
+        String issuer = "https://sample.okta.com"
```
Comment (JayNewstrom) [inline impl/src/test/groovy/com/okta/idx/sdk/impl/util/TestClientUtil.groovy:52]: Can we add another test for trailing slashes?

**Review by JayNewstrom — APPROVED** [review]
Looks good!

File: api/src/main/java/com/okta/idx/sdk/api/util/ClientUtil.java (hunk 1 above)
Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/util/ClientUtil.java:2]: done

File: api/src/main/java/com/okta/idx/sdk/api/util/ClientUtil.java (hunk 2 above)
Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/util/ClientUtil.java:43]: yeah, good point.. updated.

File: impl/src/test/groovy/com/okta/idx/sdk/impl/util/TestClientUtil.groovy (hunk 3 above)
Comment (arvindkrishnakumar-okta) [reply impl/src/test/groovy/com/okta/idx/sdk/impl/util/TestClientUtil.groovy:52]: below lines test that already.

image

**Review by JayNewstrom — APPROVED** [review]

**Review by JayNewstrom — APPROVED** [review]

**Pull request merged**